app/services/video_conversion/aws_media_convert_adapter.rb

```diff
@@ -0,0 +1,267 @@
+      rescue Discourse::InvalidParameters => e
+        Rails.logger.error("Invalid parameters for upload #{@upload.id}: #{e.message}")
+        false
+      rescue => e
+        Rails.logger.error(
+          "Unexpected error in video conversion for upload #{@upload.id}: #{e.class.name} - #{e.message}",
+        )
+        Discourse.warn_exception(
+          e,
+          message: "Unexpected error in video conversion",
+          env: {
+            upload_id: @upload.id,
+          },
+        )
+        false
+      end
+    end
+
+    def check_status(job_id)
+      response = mediaconvert_client.get_job(id: job_id)
+
+      case response.job.status
+      when "COMPLETE"
+        :complete
+      when "ERROR"
+        Rails.logger.error("MediaConvert job #{job_id} failed")
+        :error
+      when "SUBMITTED", "PROGRESSING"
+        :pending
+      else
+        Rails.logger.warn(
+          "Unexpected MediaConvert job status for job #{job_id}: #{response.job.status}",
+        )
+        :error
+      end
+    end
+
+    def handle_completion(job_id, output_path, new_sha1)
+      s3_store = FileStore::S3Store.new
+      path = "#{output_path}.mp4"
+      object = s3_store.object_from_path(path)
+
+      return false unless object&.exists?
+
+      begin
+        optimized_video =
+          create_optimized_video_record(
+            output_path,
+            new_sha1,
+            object.size,
+            "//#{s3_store.s3_bucket}.s3.dualstack.#{SiteSetting.s3_region}.amazonaws.com/#{path}",
+          )
+
+        if optimized_video
+          update_posts_with_optimized_video
+          true
+        else
+          Rails.logger.error("Failed to create OptimizedVideo record for upload #{@upload.id}")
+          false
+        end
+      rescue => e
+        Rails.logger.error(
+          "Error processing video completion for upload #{@upload.id}: #{e.message}",
+        )
+        Discourse.warn_exception(
+          e,
+          message: "Error in video processing completion",
+          env: {
+            upload_id: @upload.id,
+            job_id: job_id,
+          },
+        )
+        false
+      end
+    end
+
+    private
+
+    def valid_settings?
+      SiteSetting.video_conversion_enabled && SiteSetting.mediaconvert_role_arn.present? &&
+        SiteSetting.mediaconvert_endpoint.present?
+    end
+
+    def mediaconvert_client
+      @mediaconvert_client ||=
+        begin
+          # For some reason the endpoint is not visible in the aws console UI so we need to get it from the API
+          if SiteSetting.mediaconvert_endpoint.blank?
+            client =
+              Aws::MediaConvert::Client.new(
+                region: SiteSetting.s3_region,
+                credentials:
+                  Aws::Credentials.new(
+                    SiteSetting.s3_access_key_id,
+                    SiteSetting.s3_secret_access_key,
+                  ),
+              )
+            resp = client.describe_endpoints
```

Comment:
Is it possible this could fail with an error? If so we should handle it.

It would be good if we could do this as soon as you enabled the site setting, but I don't think we have a good mechanism for that 🤔 Otherwise this could silently fail and the conversion would never work, and admins wouldn't know.

One possible thing to do is  make an admin problem check that checks if this exists with the current settings, and if not tell admins they have to fix up the keys/permissions and so on